[ServiceBus] improve memory usage of service bus client (#19915)
Azure/azure-sdk-for-python: sdk/servicebus/azure-servicebus/CHANGELOG.md

### Other Changes

> - Improved memory usage of `ServiceBusClient` to automatically de-registered spawned `ServiceBusSender` and `ServiceBusReceiver` which are closed.

Reviewer: small nit, flow:
```suggestion
- Improved memory usage of `ServiceBusClient` by automatically de-registering spawned `ServiceBusSender` and `ServiceBusReceiver` that are closed.
```
